mathesar/filters.py

```diff
@@ -3,6 +3,16 @@
 
 from mathesar.models import Schema, Table, Database
 
+FILTER_OPTIONS_BY_TYPE_NAME = {
+    'Boolean':
+        {
+            'eq': '<boolean>',
```

Comment:
I think it might be clearer to have a more descriptive format. I was thinking something like:

```
{
  'op': 'eq',
  'allowed_values': ['boolean'],
  'num_values': 1,
  'description': 'This operation checks for exact equality'
}
```

or

```
{
  'op': 'is_null',
  'allowed_values': ['boolean', None],
  'num_values': 1,
  'description': 'This operation checks for null values'
}
```

`allowed_values` will allow us to specify the values that are acceptable. `num_values` will allow us to support filters that require multiple inputs (such as "between"). I think the description will be useful for API consumers to understand the filter.

This format also meets the "The portion of the API response describing to a given resource should always contain the same set of keys." standard from our [API standards](https://wiki.mathesar.org/en/engineering/architecture/api-standards).

Reply:
Also, I think in this case it would be useful to have a helper dictionary mapping `description` to `op` so that we don't have to repeat the same description for the same `op` across different data types.

Reply:
Thanks for this input! I wasn't sure exactly what that dictionary should look like.